doc/adr/0013-flame-graph-support.md

```diff
@@ -0,0 +1,132 @@
+- To get the tooltip of state, the FE state needs to have pointers to the data tree entry (row) to be able query the back-end  
+
+#### Visualization
+
+To visualize the data weighted `DATA_TREE` data as flame graph in the theia-trace-extesion, a new react commponent should be implemented either using timeline-chart library (similar to the `TimegraphOutputComponent`). Alternatively, the new react component can implement or use an existing flame graph library, for example [d3-flame-graph](https://www.npmjs.com/package/d3-flame-graph/.
+
+### Other considerations
+
+#### Custom Aggregation
+Flame graphs can have different schemes for aggregation of layers, e.g. per process ID, threads or full aggregation, which will change the tree and duration values of each segment. Changing the aggregation scheme needs to be part of the TSP protocol, i.e. the flame graph data provider needs to provide the available aggregation schemes and there needs to be a query parameter or path parameter to specify which one to apply in the back-end.
+
+The data provider descripter capabilities can be augmented to indicate if a data tree can change the aggregation schem. With this in place we can define how the possible groupings are provide by the server to the client, and how the client provides the aggregation scheme. 
+
+Aggregation will have to be supported in either of the solutions above.
+
+#### Weight metric
+
+Please note that the weight metric of the flame graph and weighted tree in the description above is duration. However, the weight metric can be another metric, e.g. number of calls, bytes allocated, number of page faults. The solution should allow those use cases. To support that there needs to be a description of the metric used so that the visualization is adapted accordingly (e.g. show the correct units).
+
+#### Actions
+
+The Flame Graph in Trace Comapass has built in statisics, that allows to navigate to the time range of the minimum or maximum duration in the trace. For solution one, the statistics are shown as part of the segment tooltip which is queried on demand. The min/max time range has to be encoded in the tooltip and the UI needs to be able to understand that this is an action. For that we need to define a action indicator (e.g. prefix #ACTION or similar). 
+
+For the data tree solution, the statistics could be provided as tooltip on the row as well. For that a new tooltip endpoint endpoint for data trees is required. The action design would be the same as in solution 1.
+
+## Decision
+
+Flame Graph support will be added. Re-using the flame graph data provider (solution 1) will allow client FE to leverage existing gantt chart view implementation. It also provides known API.
```

Comment:
Is it just a time graph that isn't synchronized with the global view range? Maybe we could keep using the same timegraph endpoint but add a new capability that indicates whether the data provider is dependent on global view range? Unfortunately though, everywhere in the chain 'time' is used as implied x-unit.